controller.py

```diff
@@ -359,23 +363,18 @@ def _make_request(url, on_404, on_timeout, on_exception, allow_401=False):
             return INVALID_INTEGRATION_DOCUMENT.detailed(failure_detail)
 
         # Make sure the authentication document includes a way for
-        # patrons to get help or file a copyright complaint.
-        links_by_rel = defaultdict(list)
-        for l in auth_document.links:
-            links_by_rel[l.get('rel')].append(l)
+        # patrons to get help or file a copyright complaint. Store these
+        # links in the database as Hyperlink objects.
+        links = auth_document.links or []
         for rel, problem_title in [
-                ('help', "No valid patron help email address"),
-                ("http://librarysimplified.org/rel/designated-agent/copyright",
-                 "No valid copyright designated agent email address")
+            ('help', "Invalid or missing patron support email address"),
+            (Hyperlink.COPYRIGHT_DESIGNATED_AGENT_REL,
+             "Invalid or missing copyright designated agent email")
```

Comment:
I changed the language here to match the language in the circulation manager. I need to make a few small changes to the circulation manager, which I'll put into my next circulation branch.